`Loader/classloader/InlineArray/InlineArrayValid.cs` need to be updated to work with latest roslyn

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=339983
Build error leg or test failing: Build / windows-x64 Checked NativeAOT / Build Tests
Pull request: https://github.com/dotnet/runtime/pull/88188
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

```
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(184,28): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(185,28): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(201,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(202,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(209,36): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(210,40): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(232,40): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(233,44): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(245,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(246,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(253,36): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(254,40): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(259,20): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(260,20): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(294,29): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(294,29): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(295,31): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(295,30): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(309,14): error CS0306: The type 'Span<object>' may not be used as a type argument [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(309,13): error CS0193: The * or -> operator must be applied to a pointer [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(429,43): error CS1061: '(object, short)' does not contain a definition for 'o' and no accessible extension method 'o' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs(430,47): error CS1061: '(object, short)' does not contain a definition for 's' and no accessible extension method 's' accepting a first argument of type '(object, short)' could be found (are you missing a using directive or an assembly reference?) [/__w/1/s/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj] [/__w/1/s/src/tests/build.proj]
```

and @AlekseyTs's [response](https://github.com/dotnet/runtime/pull/88188#issuecomment-1636636617]):
> I looked at several errors, all of them are expected. Compiler no longer uses user-defined indexers for indexing into inline array types.

> For `"error CS1061: '(object, short)' does not contain a definition for 'o' ..."`. Definition of the element field doesn't include tuple element names, but that is the element type that language is using. To fix, add element names to the declaration of the element field. I think custom indexer can be safely deleted.

> For `"error CS0306: The type 'Span' may not be used as a type argument"`. Language doesn't support accessing elements with types that cannot be elements of a span type themselves. A possible workaround is to change indexer to a method and use that to access elements instead of using element access expression syntax for such inline arrays.

<!--Known issue error report start -->
### Report
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|
<!--Known issue error report end -->

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.cs
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.cs
@@ -10,14 +10,6 @@ using Xunit;
 
 unsafe class Validate
 {
-    [InlineArray(1)]
-    [StructLayout(LayoutKind.Explicit)]
-    private struct Explicit
-    {
-        [FieldOffset(0)]
-        public Guid Guid;
-    }
-
     [Fact]
     public static void Explicit_Fails()
     {
@@ -28,12 +20,6 @@ unsafe class Validate
         {
             return sizeof(Explicit);
         });
-    }
-
-    [InlineArray(0)]
-    private struct ZeroLength
-    {
-        public int field;
     }
 
     [Fact]
@@ -74,12 +60,6 @@ unsafe class Validate
         });
     }
 
-    [InlineArray(-1)]
-    private struct NegativeLength
-    {
-        public long field;
-    }
-
     [Fact]
     public static void NegativeLength_Fails()
     {
@@ -96,13 +76,6 @@ unsafe class Validate
         });
     }
 
-
-    [InlineArray(123)]
-    private struct NoFields
-    {
-        public static int x;
-    }
-
     [Fact]
     public static void NoFields_Fails()
     {
@@ -113,13 +86,6 @@ unsafe class Validate
         {
             return (new NoFields()).ToString();
         });
-    }
-
-    [InlineArray(1)]
-    private struct TwoFields
-    {
-        int a;
-        int b;
     }
 
     [Fact]

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.csproj
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayInvalid.csproj
@@ -2,13 +2,12 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
-    <!-- Issue: https://github.com/dotnet/runtime/issues/88861 -->
-    <DisableProjectBuild>true</DisableProjectBuild>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="InlineArrayInvalid.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="InvalidCSharp.ilproj" />
     <ProjectReference Include="$(TestSourceDir)Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj" />
   </ItemGroup>
 </Project>

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
@@ -23,15 +23,6 @@ struct MyArray<T> : IEnumerable<T>
     public int Length => LengthConst;
 
     [UnscopedRef]
-    public ref T At(int i)
-    {
-        if ((uint)i >= (uint)Length)
-            throw new IndexOutOfRangeException(nameof(i));
-
-        return ref Unsafe.Add(ref _element, i);
-    }
-
-    [UnscopedRef]
     public Span<T> AsSpan() => MemoryMarshal.CreateSpan<T>(ref _element, Length);
 
     IEnumerator IEnumerable.GetEnumerator() => (IEnumerator<T>)this.GetEnumerator();
@@ -104,9 +95,6 @@ unsafe class Validate
     {
         public const int Length = 42;
         public E e;
-
-        [UnscopedRef]
-        public ref E At(int i) => ref Unsafe.Add(ref e, i);
     }
 
     static object s;
@@ -165,10 +153,7 @@ unsafe class Validate
     struct ObjShortArr
     {
         public const int Length = 100;
-        public (object, short) element;
-
-        [UnscopedRef]
-        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
+        public (object o, short s) element;
 
         [MethodImpl(MethodImplOptions.NoInlining)]
         public static ObjShortArr CreateArray(int recCount) {
@@ -178,8 +163,8 @@ unsafe class Validate
                 var arr = new ObjShortArr();
                 for (short i = 0; i < ObjShortArr.Length; i++)
                 {
-                    arr.At(i).o = i;
-                    arr.At(i).s = (short)(i + 1);
+                    arr[i].o = i;
+                    arr[i].s = (short)(i + 1);
                 }
                 return arr;
             }
@@ -195,16 +180,16 @@ unsafe class Validate
         var arr = new ObjShortArr();
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            arr.At(i).o = i;
-            arr.At(i).s = (short)(i + 1);
+            arr[i].o = i;
+            arr[i].s = (short)(i + 1);
         }
 
         GC.Collect(2, GCCollectionMode.Forced, true, true);
 
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            Assert.Equal(i, arr.At(i).o);
-            Assert.Equal(i + 1, arr.At(i).s);
+            Assert.Equal(i, arr[i].o);
+            Assert.Equal(i + 1, arr[i].s);
         }
     }
 
@@ -213,10 +198,7 @@ unsafe class Validate
     ref struct ObjShortArrRef
     {
         public const int Length = 100;
-        public (object, short) element;
-
-        [UnscopedRef]
-        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
+        public (object o, short s) element;
     }
 
     [MethodImpl(MethodImplOptions.NoInlining)]
@@ -226,8 +208,8 @@ unsafe class Validate
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i * 2, arr.At(i).o);
-            Assert.Equal(i * 2 + 1, arr.At(i).s);
+            Assert.Equal(i * 2, arr[i].o);
+            Assert.Equal(i * 2 + 1, arr[i].s);
         }
     }
 
@@ -239,22 +221,22 @@ unsafe class Validate
         var arr = new ObjShortArrRef();
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr.At(i).o = i;
-            arr.At(i).s = (short)(i + 1);
+            arr[i].o = i;
+            arr[i].s = (short)(i + 1);
         }
 
         GC.Collect(2, GCCollectionMode.Forced, true, true);
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i, arr.At(i).o);
-            Assert.Equal(i + 1, arr.At(i).s);
+            Assert.Equal(i, arr[i].o);
+            Assert.Equal(i + 1, arr[i].s);
         }
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr.At(i).o = i * 2;
-            arr.At(i).s = (short)(i * 2 + 1);
+            arr[i].o = i * 2;
+            arr[i].s = (short)(i * 2 + 1);
         }
 
         TestRefLikeOuterMethodArg(arr);
@@ -351,7 +333,7 @@ unsafe class Validate
         MyArray<object> arr = default;
         for (int i = 0; i < arr.Length; i++)
         {
-            arr.At(i) = i;
+            arr[i] = i;
         }
 
         BoxedMethodArg(arr);
@@ -403,8 +385,8 @@ unsafe class Validate
 
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            Assert.Equal(i, holder.arr.At(i).o);
-            Assert.Equal(i + 1, holder.arr.At(i).s);
+            Assert.Equal(i, holder.arr[i].o);
+            Assert.Equal(i + 1, holder.arr[i].s);
         }
     }
 }

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
@@ -23,15 +23,12 @@ struct MyArray<T> : IEnumerable<T>
     public int Length => LengthConst;
 
     [UnscopedRef]
-    public ref T this[int i]
+    public ref T At(int i)
     {
-        get
-        {
-            if ((uint)i >= (uint)Length)
-                throw new IndexOutOfRangeException(nameof(i));
+        if ((uint)i >= (uint)Length)
+            throw new IndexOutOfRangeException(nameof(i));
 
-            return ref Unsafe.Add(ref _element, i);
-        }
+        return ref Unsafe.Add(ref _element, i);
     }
 
     [UnscopedRef]
@@ -109,7 +106,7 @@ unsafe class Validate
         public E e;
 
         [UnscopedRef]
-        public ref E this[int i] => ref Unsafe.Add(ref e, i);
+        public ref E At(int i) => ref Unsafe.Add(ref e, i);
     }
 
     static object s;
@@ -171,7 +168,7 @@ unsafe class Validate
         public (object, short) element;
 
         [UnscopedRef]
-        public ref (object o, short s) this[int i] => ref Unsafe.Add(ref element, i);
+        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
 
         [MethodImpl(MethodImplOptions.NoInlining)]
         public static ObjShortArr CreateArray(int recCount) {
@@ -181,8 +178,8 @@ unsafe class Validate
                 var arr = new ObjShortArr();
                 for (short i = 0; i < ObjShortArr.Length; i++)
                 {
-                    arr[i].o = i;
-                    arr[i].s = (short)(i + 1);
+                    arr.At(i).o = i;
+                    arr.At(i).s = (short)(i + 1);
                 }
                 return arr;
             }
@@ -198,16 +195,16 @@ unsafe class Validate
         var arr = new ObjShortArr();
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            arr[i].o = i;
-            arr[i].s = (short)(i + 1);
+            arr.At(i).o = i;
+            arr.At(i).s = (short)(i + 1);
         }
 
         GC.Collect(2, GCCollectionMode.Forced, true, true);
 
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            Assert.Equal(i, arr[i].o);
-            Assert.Equal(i + 1, arr[i].s);
+            Assert.Equal(i, arr.At(i).o);
+            Assert.Equal(i + 1, arr.At(i).s);
         }
     }
 
@@ -219,7 +216,7 @@ unsafe class Validate
         public (object, short) element;
 
         [UnscopedRef]
-        public ref (object o, short s) this[int i] => ref Unsafe.Add(ref element, i);
+        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
     }
 
     [MethodImpl(MethodImplOptions.NoInlining)]
@@ -229,8 +226,8 @@ unsafe class Validate
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i * 2, arr[i].o);
-            Assert.Equal(i * 2 + 1, arr[i].s);
+            Assert.Equal(i * 2, arr.At(i).o);
+            Assert.Equal(i * 2 + 1, arr.At(i).s);
         }
     }
 
@@ -242,48 +239,28 @@ unsafe class Validate
         var arr = new ObjShortArrRef();
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr[i].o = i;
-            arr[i].s = (short)(i + 1);
+            arr.At(i).o = i;
+            arr.At(i).s = (short)(i + 1);
         }
 
         GC.Collect(2, GCCollectionMode.Forced, true, true);
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i, arr[i].o);
-            Assert.Equal(i + 1, arr[i].s);
+            Assert.Equal(i, arr.At(i).o);
+            Assert.Equal(i + 1, arr.At(i).s);
         }
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr[i].o = i * 2;
-            arr[i].s = (short)(i * 2 + 1);
+            arr.At(i).o = i * 2;
+            arr.At(i).s = (short)(i * 2 + 1);
         }
 
         TestRefLikeOuterMethodArg(arr);
     }
 
     // ====================== RefLikeInner ========================================================
-    [InlineArray(LengthConst)]
-    ref struct SpanArr
-    {
-        private const int LengthConst = 100;
-        public Span<object> element;
-
-        public Span<object>* this[int i]
-        {
-            get
-            {
-                fixed (Span<object>* p = &element)
-                {
-                    return p + i;
-                }
-            }
-        }
-
-        public int Length => LengthConst;
-    }
-
     [MethodImpl(MethodImplOptions.NoInlining)]
     static void TestRefLikeInnerMethodArg(SpanArr arr)
     {
@@ -291,8 +268,8 @@ unsafe class Validate
 
         for (int i = 1; i < arr.Length; i++)
         {
-            Assert.Equal(i, arr[i]->Length);
-            Assert.Equal(i, (*arr[i])[0]);
+            Assert.Equal(i, arr.At(i)->Length);
+            Assert.Equal(i, (*arr.At(i))[0]);
         }
     }
 
@@ -306,7 +283,7 @@ unsafe class Validate
         {
             var objArr = new object[i];
             objArr[0] = i;
-            *arr[i] = objArr;
+            *arr.At(i) = objArr;
         }
 
         TestRefLikeInnerMethodArg(arr);
@@ -374,7 +351,7 @@ unsafe class Validate
         MyArray<object> arr = default;
         for (int i = 0; i < arr.Length; i++)
         {
-            arr[i] = i;
+            arr.At(i) = i;
         }
 
         BoxedMethodArg(arr);
@@ -426,8 +403,8 @@ unsafe class Validate
 
         for (short i = 0; i < ObjShortArr.Length; i++)
         {
-            Assert.Equal(i, holder.arr[i].o);
-            Assert.Equal(i + 1, holder.arr[i].s);
+            Assert.Equal(i, holder.arr.At(i).o);
+            Assert.Equal(i + 1, holder.arr.At(i).s);
         }
     }
 }

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayValid.cs
@@ -198,7 +198,10 @@ unsafe class Validate
     ref struct ObjShortArrRef
     {
         public const int Length = 100;
-        public (object o, short s) element;
+        public (object, short) element;
+
+        [UnscopedRef]
+        public ref (object o, short s) At(int i) => ref Unsafe.Add(ref element, i);
     }
 
     [MethodImpl(MethodImplOptions.NoInlining)]
@@ -208,8 +211,8 @@ unsafe class Validate
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i * 2, arr[i].o);
-            Assert.Equal(i * 2 + 1, arr[i].s);
+            Assert.Equal(i * 2, arr.At(i).o);
+            Assert.Equal(i * 2 + 1, arr.At(i).s);
         }
     }
 
@@ -221,22 +224,22 @@ unsafe class Validate
         var arr = new ObjShortArrRef();
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr[i].o = i;
-            arr[i].s = (short)(i + 1);
+            arr.At(i).o = i;
+            arr.At(i).s = (short)(i + 1);
         }
 
         GC.Collect(2, GCCollectionMode.Forced, true, true);
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            Assert.Equal(i, arr[i].o);
-            Assert.Equal(i + 1, arr[i].s);
+            Assert.Equal(i, arr.At(i).o);
+            Assert.Equal(i + 1, arr.At(i).s);
         }
 
         for (short i = 0; i < ObjShortArrRef.Length; i++)
         {
-            arr[i].o = i * 2;
-            arr[i].s = (short)(i * 2 + 1);
+            arr.At(i).o = i * 2;
+            arr.At(i).s = (short)(i * 2 + 1);
         }
 
         TestRefLikeOuterMethodArg(arr);

--- a/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj
+++ b/src/tests/Loader/classloader/InlineArray/InlineArrayValid.csproj
@@ -2,13 +2,12 @@
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <OutputType>Exe</OutputType>
-    <!-- Issue: https://github.com/dotnet/runtime/issues/88861 -->
-    <DisableProjectBuild>true</DisableProjectBuild>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="InlineArrayValid.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="InvalidCSharp.ilproj" />
     <ProjectReference Include="$(TestSourceDir)Common/CoreCLRTestLibrary/CoreCLRTestLibrary.csproj" />
   </ItemGroup>
 </Project>

--- a/src/tests/Loader/classloader/InlineArray/InvalidCSharp.il
+++ b/src/tests/Loader/classloader/InlineArray/InvalidCSharp.il
@@ -1,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+.assembly extern System.Runtime { .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A ) }
+
+.assembly InvalidCSharp { }
+
+.class public explicit ansi sealed beforefieldinit Explicit
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 01 00 00 00 00 00
+    )
+    .field [0] public valuetype [System.Runtime]System.Guid Guid
+}
+
+.class public sequential ansi sealed beforefieldinit ZeroLength
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 00 00 00 00 00 00
+    )
+    .field public int32 'field'
+}
+
+.class public sequential ansi sealed beforefieldinit TooLarge
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 00 00 00 20 00 00
+    )
+    .field public int64 'field'
+}
+
+.class public sequential ansi sealed beforefieldinit NegativeLength
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 ff ff ff ff 00 00
+    )
+    .field public int64 'field'
+}
+
+.class public sequential ansi sealed beforefieldinit NoFields
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 7b 00 00 00 00 00
+    )
+    .pack 0
+    .size 1
+    .field public static int32 x
+}
+
+.class public sequential ansi sealed beforefieldinit TwoFields
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 01 00 00 00 00 00
+    )
+    .field private int32 a
+    .field private int32 b
+}

--- a/src/tests/Loader/classloader/InlineArray/InvalidCSharp.il
+++ b/src/tests/Loader/classloader/InlineArray/InvalidCSharp.il
@@ -61,3 +61,50 @@
     .field private int32 a
     .field private int32 b
 }
+
+.class public sequential ansi sealed beforefieldinit SpanArr
+    extends [System.Runtime]System.ValueType
+{
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.IsByRefLikeAttribute::.ctor() = (
+        01 00 00 00
+    )
+    .custom instance void [System.Runtime]System.Runtime.CompilerServices.InlineArrayAttribute::.ctor(int32) = (
+        01 00 64 00 00 00 00 00
+    )
+
+    .field private static literal int32 LengthConst = int32(100)
+    .field public valuetype [System.Runtime]System.Span`1<object> element
+
+    .method public hidebysig
+        instance valuetype [System.Runtime]System.Span`1<object>* At (
+            int32
+        ) cil managed
+    {
+        .locals init (
+            [0] valuetype [System.Runtime]System.Span`1<object>& pinned
+        )
+        ldarg.0
+        ldflda valuetype [System.Runtime]System.Span`1<object> SpanArr::element
+        stloc.0
+        ldloc.0
+        conv.u
+        ldarg.1
+        conv.i
+        sizeof valuetype [System.Runtime]System.Span`1<object>
+        mul
+        add
+        ret
+    }
+
+    .method public hidebysig specialname
+        instance int32 get_Length () cil managed
+    {
+        ldc.i4.s 100
+        ret
+    }
+
+    .property instance int32 Length()
+    {
+        .get instance int32 SpanArr::get_Length()
+    }
+}

--- a/src/tests/Loader/classloader/InlineArray/InvalidCSharp.ilproj
+++ b/src/tests/Loader/classloader/InlineArray/InvalidCSharp.ilproj
@@ -1,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk.IL">
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="InvalidCSharp.il" />
+  </ItemGroup>
+</Project>

